### ISSUE TITLE
Data does not appear to be streaming to device

### ISSUE THREAD

**Issue by lightswitch2159** [NimmLor/IoT-Audio-Visualization-Center#6]
Hello!

I've tried everything that I can think of, but no matter what, no data seems to be getting to my LED device. If I enable a visualizer preset and turn data streaming on, no effect is seen. I see the waveform like it should be working, but nothing is actually happening.

**Comment by lazaq** [comment]
It doesn't work for me either. I can control the brightness or turn it on and off. but the music data doesn't work

**Comment by L3-N0X** [comment]
Same with my Nanoleafes...

**Comment by NimmLor** [comment]
Yeah, we already fixed that issue but we haven't drafted a new release yet. Please download the newest version of the `in-development` branch [here](https://github.com/NimmLor/esp8266-fastled-iot-webserver/tree/in-development).

**Comment by L3-N0X** [comment]
Okay, thanks for the quick response, I will download it right now! Will sure be epic! Thanks for this nice project!

**Comment by L3-N0X** [comment]
So, I downloaded the indev-Version, but now I can´t select the Visualize Patterns... Is there a way to get these patterns again or do I have to wait for the next release?

**Comment by NimmLor** [comment]
You have probably forgot to enable `#define ENABLE_UDP_VISUALIZATION `, it was moved down there because the majority isn't using this feature.

**Comment by L3-N0X** [comment]
Oh yes, I´ve not seen this... Thanks!
I´ve tested it now and it works!

**Comment by L3-N0X** [comment]
I have a question: I want to change the bullet´s Size, but when i enter an other value in Code Line 3526 (`#define BULLET_COLOR_SIZE`), nothing changes. I want to change it, because on my Nanoleafes the Bullet is 4 Leafes big. I have 9 Leafes á 6 LEDs... Is there a way to change it or did I just made a mistake anywhere?
Du kannst btw auch auf Deutsch antworten wenn du willst 😄

**Comment by freddiN** [comment]
Thanks for the hints in here, I managed to get it working by using the in-dev branch and defining ENABLE_UDP_VISUALIZATION.

It doenst seem to be in sync with the music though, is it for you? I mean I see a reaction, like when the music gets quiter the visualization tunes it down after a few seconds, but its like a 5 second delay ... I already played around with the smoothing and speed settings.

Otherwise my Wemos D1 might be too slow to keep up, or my fritzbox isnt UDPing fast enough :)

All in all a big thumbs up Lorenz, the whole project is simply awesome, from the STLs on thingiverse to the software part over here at github. Good job!

**Comment by L3-N0X** [comment]
My Nanoleafs are sync to the music, which Pattern do you use and what Volume? I had to put my Output of my headset in the Analyzer up to 200% that it looked cool and I use right now the rainbow band visualizer. With the right song, it looks also better. 

But if you're fritzbox is the problem, I don't know what to do. 

Yes, the project is very awesome!

**Comment by freddiN** [comment]
It looks stuttery and delayed for me:_
https://streamable.com/p76tlc
But I know its doing something, it becomes dark when music fades out...

My settings:
![screen_1](https://user-images.githubusercontent.com/14030572/106742243-12c64300-661d-11eb-9e76-989c81cb4c27.JPG)
![screen_2](https://user-images.githubusercontent.com/14030572/106742251-15289d00-661d-11eb-9716-5340c9a0d06d.JPG)
![screen_3](https://user-images.githubusercontent.com/14030572/106742255-16f26080-661d-11eb-8ddd-80dad23f4a82.JPG)
![screen_4](https://user-images.githubusercontent.com/14030572/106742260-1954ba80-661d-11eb-8bd7-20ca9999ecef.JPG)

Any ideas?

What I tried so far:
- set LED_DEBUG 0 to switch off debug messages
- play around with all settings in the windows app
- chose all visualization patterns on Wemos
- "clean flash" (Erase flash: "all contents" in Arduino)

If its not one of the settings, then I suspect its either the Wemos or my network.
To see if its my network I would like to directly connect to the Wemos, is that still possible? Putting it in AP mode? Because since the switch to Wifimanager it seems like I cannot do that any more ... after a clean flash, when it starts up its own AP, I can only go to WifiManger but not do the LED-pattern selection it seems ...

**Comment by mathbrook** [comment]
> It looks stuttery and delayed for me:_
> https://streamable.com/p76tlc
> But I know its doing something, it becomes dark when music fades out ...


I have the exact same issue, assumed its a matter of hardware or internet. I'm using a HiLetgo NodeMCU and WS2812B.

**Comment by aceofjokers** [comment]
First I want to say - Awesome work!!! Love what you've done.

Also noticed there's an issue when compiling the in-dev

line 624

missing a comma after "Single Color Visualization Outside"


```
#if LED_DEVICE_TYPE == 1                  // Matrix                          // palet  speed  color  spark  twinkle
    { RainbowVisualizer,                  "Rainbow Visualization",              true,  true,  false, false, false},
    { SingleColorVisualizer,              "Single Color Visualization",         true,  true,  true,  false, false},
    { RainbowVisualizerDoubleSided,       "Rainbow Visualization Outside",      true,  true,  false, false, false},
    { SingleColorVisualizerDoubleSided,   "Single Color Visualization Outside"  true,  true,  true,  false, false},

```

Compile log....
```
In file included from C:\Users\Dean\Documents\Arduino\esp8266-fastled-iot-webserver-4.5\esp8266-fastled-iot-webserver\esp8266-fastled-iot-webserver.ino:20:0:
C:\Users\Dean\Documents\Arduino\libraries\FastLED\src/FastLED.h:14:21: note: #pragma message: FastLED version 3.004.000
 #    pragma message "FastLED version 3.004.000"
                     ^
In file included from C:\Users\Dean\Documents\Arduino\libraries\FastLED\src/FastLED.h:65:0,
                 from C:\Users\Dean\Documents\Arduino\esp8266-fastled-iot-webserver-4.5\esp8266-fastled-iot-webserver\esp8266-fastled-iot-webserver.ino:20:
C:\Users\Dean\Documents\Arduino\libraries\FastLED\src/fastspi.h:135:23: note: #pragma message: No hardware SPI pins defined.  All SPI access will default to bitbanged output
 #      pragma message "No hardware SPI pins defined.  All SPI access will default to bitbanged output"
                       ^
In file included from C:\Users\Dean\AppData\Local\Arduino15\packages\esp8266\hardware\esp8266\2.7.4\cores\esp8266/Arduino.h:29:0,
                 from sketch\esp8266-fastled-iot-webserver.ino.cpp:1:
esp8266-fastled-iot-webserver:624:81: error: expected '}' before 'true'
     { SingleColorVisualizerDoubleSided,   "Single Color Visualization Outside"  true,  true,  true,  false, false},
                                                                                 ^
esp8266-fastled-iot-webserver:624:81: error: expected '}' before 'true'
esp8266-fastled-iot-webserver:624:81: error: expected ',' or ';' before 'true'
esp8266-fastled-iot-webserver:624:114: error: expected declaration before '}' token
     { SingleColorVisualizerDoubleSided,   "Single Color Visualization Outside"  true,  true,  true,  false, false},
                                                                                                                  ^
exit status 1
expected '}' before 'true'
```
